tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -89,24 +121,85 @@ spec:
         echo -n $commonTags > $(results.commonTags.path)
 
         application=$(jq -r '.application' "${SNAPSHOT_SPEC_FILE}")
+        NUM_COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
         printf 'Beginning "%s" for "%s"\n\n' "$(context.task.name)" "$application"
-        for component in $(jq -rc '.components[]' "${SNAPSHOT_SPEC_FILE}")
+        for ((i = 0; i < $NUM_COMPONENTS; i++))
         do
+          component=$(jq -c --argjson i "$i" '.components[$i]' "${SNAPSHOT_SPEC_FILE}")
           containerImage=$(jq -r '.containerImage' <<< $component)
           repository=$(jq -r '.repository' <<< $component)
+          imageTags=$(jq '.tags' <<< $component)
 
           # Just read the first from the list of architectures
           read -r arch_json <<< $(get-image-architectures "${containerImage}")
           arch=$(echo "${arch_json}" | jq -r .platform.architecture)
 
           name=$(jq -r '.name' <<< $component)
           git_sha=$(jq -r '.source.git.revision' <<< $component) # this sets the value to "null" if it doesn't exist
+          build_sha=$(echo "${containerImage}" | cut -d ':' -f 2)
+          passedTimestampFormat=$(jq -r --arg default $defaultTimestampFormat \
+            '.timestampFormat // $default' <<< $component)
+          timestamp="$(date "+$passedTimestampFormat")"
+
+          allTagsPreSubstitution=$(jq -n --argjson defaults "$defaultTags" --argjson imageTags \
+            "$imageTags" '$defaults? + $imageTags? | unique')
+          tags=$(translate_tags "${allTagsPreSubstitution}" "${timestamp}" "${git_sha}" \
+            "${git_sha:0:7}" "${build_sha}")
```

Comment:
Updated the README to say it is first 7 chars. That is the short git sha tag we already have in the task so I want to keep it the same. I think it would be way nicer to let them trim to any digit they want, but I don't have a great solution for that and don't want to hold up this PR with too much logic for just one tag

Reply:
yeah, ok, thanks